Name resolution 2.0 impl block not resolved

I tried this code:

```rust
struct Test;

impl Test {}
```

I expected to see this happen: Compiles with a few warnings (unused struct).

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ff15ab3e54dafb6bae44b9470d94d768

Instead, this happened:

```
name.rs:3:6: error: unknown reference for resolved name: ‘Test’
    3 | impl Test {}
      |      ^~~~
```

### Meta

- 2dff71f6fcb

--- a/gcc/rust/resolve/rust-default-resolver.cc
+++ b/gcc/rust/resolve/rust-default-resolver.cc
@@ -115,6 +115,7 @@ void
 DefaultResolver::visit (AST::InherentImpl &impl)
 {
   auto inner_fn = [this, &impl] () {
+    visit (impl.get_type ());
     for (auto &item : impl.get_impl_items ())
       item->accept_vis (*this);
   };

--- a/gcc/testsuite/rust/compile/name_resolution25.rs
+++ b/gcc/testsuite/rust/compile/name_resolution25.rs
@@ -1,0 +1,5 @@
+// { dg-options "-frust-name-resolution-2.0" }
+
+struct Test; // { dg-warning "struct is never constructed: .Test." }
+
+impl Test {}

